stable/grafana/values.yaml

```diff
@@ -1,475 +1,148 @@
+  path: /
+  hosts:
+    - chart-example.local
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
+resources: {}
+#  limits:
+#    cpu: 100m
+#    memory: 128Mi
+#  requests:
+#    cpu: 100m
+#    memory: 128Mi
+
+## Node labels for pod assignment
+## ref: https://kubernetes.io/docs/user-guide/node-selection/
+#
+nodeSelector: {}
+
+## Tolerations for pod assignment
+## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+##
+tolerations: []
 
-    [log.console]
-    ;level =
-    ;format = console
+## Affinity for pod assignment
+## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+##
+affinity: {}
 
-    [event_publisher]
-    ;enabled = false
-    ;rabbitmq_url = amqp://localhost/
-    ;exchange = grafana_events
+## Enable persistence using Persistent Volume Claims
+## ref: http://kubernetes.io/docs/user-guide/persistent-volumes/
+##
+persistence:
+  enabled: false
+  # storageClass: default
+  # accessModes:
+  #   - ReadWriteOnce
+  # size: 10Gi
+  # annotations: {}
+  # subPath: ""
+  # existingClaim:
 
-    [dashboards.json]
-    enabled = true
-    path = /var/lib/grafana/dashboards
+adminUser: admin
+# adminPassword: strongpassword
 
-    [metrics]
-    ;enabled           = true
-    ;interval_seconds  = 10
 
-    ; [metrics.graphite]
-    ; address = localhost:2003
-    ; prefix = prod.grafana.%(instance_name)s.
+## Extra enviornment variables that will be pass onto deployment pods
+env: {}
 
-    [grafana_net]
-    url = https://grafana.net
+# Pass the plugins you want installed as a comma separated list.
+# plugins: ""
 
 
-## Grafana dashboard files ConfigMap entries
-## If you'd like to preinstall prometheus dashboard on the same namespace as example, get it from:
-##
-## https://grafana.net/dashboards/2
+## Configure grafana datasources
+## ref: http://docs.grafana.org/administration/provisioning/#datasources
 ##
-## and add it below.
-##
-serverDashboardFiles: {}
-
-## Grafana dashboard files to import ConfigMap entries
-## The behaviuor is different then 'serverDashboardFiles' in that, this flow will create a Job to import using the API.
-## If you'd like to preinstall prometheus dashboard on the same namespace as example, get it from:
+datasources: {}
+#  datasources.yaml:
+#    apiVersion: 1
+#    datasources:
+#    - name: Prometheus
+#      type: prometheus
+#      url: http://prometheus-prometheus-server
+#      access: proxy
+#      isDefault: true
+
+## Configure grafana dashboard providers
+## ref: http://docs.grafana.org/administration/provisioning/#dashboards
 ##
-## https://grafana.net/dashboards/2
+dashboardProviders: {}
+#  dashboardproviders.yaml:
+#    apiVersion: 1
+#    providers:
+#    - name: 'default'
+#      orgId: 1
+#      folder: ''
+#      type: file
+#      disableDeletion: false
+#      editable: true
+#      options:
+#        path: /var/lib/grafana/dashboards
```

Comment:
Perhaps it's just me, but I find that I always need to provision dashboards. It would be ideal if provisioning was enabled by default, so I just needed to fill in `dashboards`

Reply:
@DaytonG  I think its a general best practice to have minimal sane defaults in values.yaml

Reply:
Yea, I can see that. My only concern is if someone tried to set `dashboards` but didn't know they _also_ needed to configure providioners it may be difficult to figure out what to do. Perhaps a comment would do for `dashboards` saying "if you need this, you must uncomment the providers configuration"?

Reply:
Added comment to values.yaml